src/Compilers/CSharp/Test/Emit3/RefStructInterfacesTests.cs

```diff
@@ -16822,7 +16822,814 @@ public class Helper
+        }
+
+        [Fact]
+        public void MissingScopedInOverride_01()
+        {
+            var src = @"
+abstract class Base
+{
+    protected abstract T Test1<T>(scoped T x)
+        where T : allows ref struct;
+
+    protected abstract S Test2(scoped S y);
+}
+
+abstract class Derived1 : Base
+{
+    protected abstract override T Test1<T>(T x);
+
+    protected abstract override S Test2(S y);
+}
+
+abstract class Derived2 : Base
+{
+    protected abstract override T Test1<T>(scoped T x);
+
+    protected abstract override S Test2(scoped S y);
+}
+
+ref struct S
+{
+}
+";
+
+            var comp = CreateCompilation(src, targetFramework: s_targetFrameworkSupportingByRefLikeGenerics);
+            comp.VerifyDiagnostics(
+                // (12,35): error CS8987: The 'scoped' modifier of parameter 'x' doesn't match overridden or implemented member.
+                //     protected abstract override T Test1<T>(T x);
+                Diagnostic(ErrorCode.ERR_ScopedMismatchInParameterOfOverrideOrImplementation, "Test1").WithArguments("x").WithLocation(12, 35),
+                // (14,35): error CS8987: The 'scoped' modifier of parameter 'y' doesn't match overridden or implemented member.
+                //     protected abstract override S Test2(S y);
+                Diagnostic(ErrorCode.ERR_ScopedMismatchInParameterOfOverrideOrImplementation, "Test2").WithArguments("y").WithLocation(14, 35)
+                );
+        }
+
+        [Fact]
+        public void MissingScopedInOverride_02()
+        {
+            var src = @"
+abstract class Base
+{
+    protected abstract void Test1<T>(scoped T x, out T z)
+        where T : allows ref struct;
+
+    protected abstract void Test2(scoped S y, out S z);
+}
+
+abstract class Derived1 : Base
+{
+    protected abstract override void Test1<T>(T x, out T z);
+
+    protected abstract override void Test2(S y, out S z);
+}
+
+abstract class Derived2 : Base
+{
+    protected abstract override void Test1<T>(scoped T x, out T z);
+
+    protected abstract override void Test2(scoped S y, out S z);
+}
+
+ref struct S
+{
+}
+";
+
+            var comp = CreateCompilation(src, targetFramework: s_targetFrameworkSupportingByRefLikeGenerics);
+            comp.VerifyDiagnostics(
+                // (12,38): error CS8987: The 'scoped' modifier of parameter 'x' doesn't match overridden or implemented member.
+                //     protected abstract override void Test1<T>(T x, out T z);
+                Diagnostic(ErrorCode.ERR_ScopedMismatchInParameterOfOverrideOrImplementation, "Test1").WithArguments("x").WithLocation(12, 38),
+                // (14,38): error CS8987: The 'scoped' modifier of parameter 'y' doesn't match overridden or implemented member.
+                //     protected abstract override void Test2(S y, out S z);
+                Diagnostic(ErrorCode.ERR_ScopedMismatchInParameterOfOverrideOrImplementation, "Test2").WithArguments("y").WithLocation(14, 38)
+                );
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Consider testing `ERR_ScopedMismatchInParameterOfPartial` and `ERR_ScopedMismatchInParameterOfTarget`.